pkgs/tools/networking/gnirehtet/default.nix

```diff
@@ -0,0 +1,47 @@
+{stdenv, rustPlatform, fetchFromGitHub, fetchzip, androidenv}:
+let
+version = "2.2.1";
+apk = stdenv.mkDerivation {
+  name = "gnirehtet.apk-${version}";
+  src = fetchzip {
+    url = "https://github.com/Genymobile/gnirehtet/releases/download/v${version}/gnirehtet-rust-linux64-v${version}.zip";
+    sha256 = "1rz2wdjc1y7n8fhskmki1nj0ak80ylxspcsrcdnjkk9r7jbq0kan";
+  };
+  installPhase = ''
+    mkdir $out
+    mv gnirehtet.apk $out
+  '';
+};
+in
+rustPlatform.buildRustPackage rec {
+  name = "gnirehtet-${version}";
+
+  src = fetchFromGitHub {
+      owner = "Genymobile";
+      repo = "gnirehtet";
+      rev = "v${version}";
+      sha256 = "1mv8nq4422k2d766qjqqnqp47qzzbbvlwhdni0k6w4nmd3m5cnd9";
+  };
+  sourceRoot = "source/relay-rust";
+  cargoSha256 = "11qf9n6h6akvb0rbmsgdlfmypkbnas8ss1cs7i8w19mh7524n0v5";
+
+  patchPhase = ''
+    substituteInPlace src/main.rs \
```

Comment:
It would be better to use an actual patch file (through `substituteAll`) so it is noticed when the substitution no longer works.

Reply:
fixed